app/controllers/shopify_app/callback_controller.rb

```diff
@@ -118,13 +122,17 @@ def set_shopify_session
       session_store = ShopifyAPI::Session.new(
         domain: shop_name,
         token: token,
-        api_version: ShopifyApp.configuration.api_version
+        api_version: ShopifyApp.configuration.api_version,
       )
 
       session[:shopify_user] = associated_user
       if session[:shopify_user].present?
         session[:shop_id] = nil if shop_session && shop_session.domain != shop_name
         session[:user_id] = ShopifyApp::SessionRepository.store_user_session(session_store, associated_user)
+        ShopifyApp::SessionRepository.clean_up_actual_sessions
+        if jwt_shopify_session_id
```

Comment:
What is our rollout strategy for this? As it stands, this would need to be a major version on `shopify_app`. Should this be opt-in for now, then opt-out? cc: @greatestape